src/main/java/ro/msg/learning/shop/dto/ViewStockDto.java

```diff
@@ -0,0 +1,14 @@
+package ro.msg.learning.shop.dto;
+
+import lombok.Builder;
+import lombok.Data;
+import ro.msg.learning.shop.domain.Location;
+import ro.msg.learning.shop.domain.Product;
+
+@Data
+@Builder
+public class ViewStockDto {
+    private Product product;
+    private Location location;
```

Comment:
Please use only one DTO: combine `ViewStockDto` with `CreateStockDto`.

For instance. you can include the following fields:

```
    private String productId;
    private String locationId;
    private Integer quantity;
```

Do not spend too much time working on the Stock controller as it was not required.